### PR TITLE
feat(db_engine): Add custom_user_agent when connecting to MotherDuck

### PR DESCRIPTION
**Pull request by guenp**
### SUMMARY
This PR adds a `custom_user_agent` key to the config that is passed to the SQLAlchemy engine for DuckDB. This is needed for MotherDuck so we can track usage of Superset.
The PR also updates the `sqlalchemy_uri_placeholder` for DuckDB and MotherDuck.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
User agent before:
```
duckdb/v0.9.2(osx_arm64) python duckdb_engine/0.11.2(sqlalchemy/1.4.36)
```

User agent after:
```
duckdb/v0.9.2(osx_arm64) python duckdb_engine/0.11.2(sqlalchemy/1.4.36) apache-superset/0.0.0-dev
```

`sqlalchemy_uri_placeholder` before:
Screenshot 2024-03-26 at 10 29 20 AM
Screenshot 2024-03-26 at 10 29 33 AM

`sqlalchemy_uri_placeholder` after:
Screenshot 2024-03-26 at 10 28 46 AM
Screenshot 2024-03-26 at 10 28 59 AM

### TESTING INSTRUCTIONS
Run unit tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/duckdb.py [hunk 1]
```diff
@@ -81,9 +84,28 @@ def get_table_names(
     ) -> set[str]:
         return set(inspector.get_table_names(schema))
 
+    @staticmethod
+    def get_extra_params(database: Database) -> dict[str, Any]:
+        """
+        Add a user agent to be used in the requests.
+        """
+        extra: dict[str, Any] = BaseEngineSpec.get_extra_params(database)
```
Comment (betodealmeida) [inline superset/db_engine_specs/duckdb.py:92]:
```suggestion
        extra: dict[str, Any] = super().get_extra_params(database)
```

**Review by betodealmeida — APPROVED** [review]
Nice!

File: superset/db_engine_specs/duckdb.py (hunk 1 above)
Comment (guenp) [reply superset/db_engine_specs/duckdb.py]: This doesn't work unfortunately because it's a `staticmethod` and there is no reference to `self`.

**Comment by guenp** [comment]
I can repro the [failing CI step](https://github.com/apache/superset/actions/runs/8452926923/job/23154446091?pr=27665#step:5:13) locally, but it succeeds if I set `flask` back to 2.2.5 (this was changed in https://github.com/apache/superset/pull/27657).
These are the errors (also happening on `main`):
```
Run pylint -j 0 superset
************* Module superset.models.helpers
superset/models/helpers.py:39:0: E06[11](https://github.com/apache/superset/actions/runs/8452926923/job/23154446091?pr=27665#step:5:12): No name 'escape' in module 'flask' (no-name-in-module)
superset/models/helpers.py:39:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.models.sql_lab
superset/models/sql_lab.py:28:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.tags.models
superset/tags/models.py:22:0: E0611: No name 'escape' in module 'flask' (no-name-in-module)
************* Module superset.views.database.mixins
superset/views/database/mixins.py:19:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.views.chart.mixin
superset/views/chart/mixin.py:[17](https://github.com/apache/superset/actions/runs/8452926923/job/23154446091?pr=27665#step:5:18):0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.connectors.sqla.views
superset/connectors/sqla/views.py:21:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.connectors.sqla.models
superset/connectors/sqla/models.py:36:0: E0611: No name 'escape' in module 'flask' (no-name-in-module)
superset/connectors/sqla/models.py:36:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
************* Module superset.utils.core
superset/utils/core.py:63:0: E0611: No name 'Markup' in module 'flask' (no-name-in-module)
```

It looks like other PRs are failing with the same issue

**Comment by guenp** [comment]
@betodealmeida Thanks for the review! this PR is now ready to merge ✅

**Review by villebro — APPROVED** [review]

**Pull request merged**